include/atomics/receiver_cadmium.hpp

```diff
@@ -22,48 +33,71 @@
 #include <limits>
 #include <random>
 
-//updated relative path --Syed Omar
 #include "../data_structures/message.hpp"
 
 using namespace cadmium;
 using namespace std;
 
-//Port definition
-// changed Receiver_defs to receiver_defs Syed Omar
+
+/** 
+* Structure Port definition for output and input messages.
+*/
 struct receiver_defs{
     struct out : public out_port<message_t> {};
     struct in : public in_port<message_t> {};
 };
-   
+
+/** 
+* The Receiver class receives message and sends out acknowledge.
+*/  
 template<typename TIME>
 class Receiver{
-    using defs = receiver_defs; // putting definitions in context
+    // putting definitions in context
+    using defs = receiver_defs; 
     public:
     //Parameters to be overwriten when instantiating the atomic model
-    TIME PREPARATION_TIME;
-    //default constructor
+    TIME PREPARATION_TIME; /**< PREPARATION_TIME is a constant that contains the time delay from acknowledge to output. */

```

Comment:
Can include brief member comment here after detailed comment and in other places:
/**< Detailed comment. */ //!<Brief comment.

Reply:
Modified